src/eradiate/radprops/_afgl1986.py

```diff
@@ -98,8 +98,8 @@ class AFGL1986RadProfile(RadProfile):
                 attrs.validators.instance_of(xr.Dataset)
             ),
         ),
-        doc="Absorption coefficient dataset. If ``None``, the absorption "
-        "coefficient is set to zero.",
+        doc="Absorption coefficient dataset. If ``has_absorption`` is "
+        " ``True``, this parameter is required. Otherwise, it is optional.",
```

Comment:
I agree this is not best. Here is the situation:
* it is difficult to set a default value to `absorption_dataset` that works whatever the spectral settings
* (as far as I understand it) we want default constructors for our scene elements, i.e. `AFGL1986RadProfile()` should not require parameters
* The default value for `has_absorption` is `True`
* (my opinion) as to `absorption_dataset`, it would be best to have the same behaviour for `AFGL1986RadProfile` and `US76ApproxRadProfile`

Given that the ongoing refactoring (support for customized atmospheric profiles) will erase all that, I did not want to spend too much time on this

Reply:
It's fine to require some arguments (see the `InterpolatedSpectrum` class, for instance).

Reply:
I'll see how to refactor this then

Reply:
> IMO the dataset should be required regardless whether absorption is off or on—otherwise it's not an AFGL atmosphere anymore.

Do you have the same opinion about `US76ApproxRadProfile`?

Reply:
I'd say yes. My view is as follows: in the basic use case, absorption modelling is active, and the switch disabling absorption is here for checks and debugging purposes. So the behaviour with `has_absorption` set to `False` is derived from the one with `has_absorption` set to `True`.

For people who just want Rayleigh absorption, we could have a `RayleighRadProfile` type, which would accept various thermophysical property profiles (obviously not part of this PR). Does this make sense to you?

Reply:
Okay I'll mark `absorption_dataset` as required parameter also for `US76ApproxRadProfile` then